### ISSUE TITLE
Randomness 'different' results in weird behavior of Dropout

### ISSUE THREAD

**Issue by link-er** [pytorch/pytorch#92283]
Hi,

I am using functorch for computing Hessian of a network with respect to parameters. In particular, I am doing it for WideResNet and I noticed, that when hessian is computed with 
```
functorch.jacfwd(functorch.jacrev(func), randomness='different')(parameters)
```
 it results in error because somehow, after dropout layer the batch dimension disappears (so all the further layers cannot be applied).
Everything works with randomness='same'

cc @ezyang @gchanan @zou3519 @Chillee @samdow @soumith @kshitij12345 @janeyx99

**Comment by zou3519** [comment]
This seems bad. Marking it high-pri for investigation, though we need to reproduce the problem. @link-er, it would help if there were a minimal script we could use to reproduce the problem, otherwise, we can try from what you've described.

**Comment by link-er** [comment]
Hi,

Thank you for reply! Here is a script for reproducing the error. One call for hessian works, while other does not.
[gist](https://gist.github.com/link-er/684c00c381a024ac09234c6bc5cca2ab)

**Comment by kshitij12345** [comment]
Minimum Repro:
```python

import torch
import torch.func
import torch.nn as nn

x = torch.randn(3, device='cuda')
y = torch.randn(1, 3, device='cuda')

def fn(x, y):
    return x + nn.functional.dropout(y, 0.3).mean(1)

# Errors for `randomness == 'different` because `mean` expects atleast 2-d tensor.
o = torch.func.vmap(fn, in_dims=(0, None), randomness='different')(x, y)

```
